src/common/spectral-utils.ts

```diff
@@ -0,0 +1,20 @@
+import { AGENT_LAST_UPDATE_DATE } from './constants'
+import { PersistenceContext } from './persistence-context'
+
+export const getLastAgentUpdateDate = (): number | undefined => {
+  return PersistenceContext.getInstance().getGlobalStateValue<number>(
+    AGENT_LAST_UPDATE_DATE
+  )
+}
```

Comment:
Consider removing this function that is in use one time in this file and move the logic to 'isOneWeekPassedSinceLastUpdate'